src/logic/transactionEngine.ts

```diff
@@ -22,32 +57,41 @@ export async function createSplTransfer(sender: string, destination: string, amo
     mint.address
   );
 
+  const amountLamports = mint.decimals === 0 ? amount : new Decimal(amount).mul(new Decimal(10).pow(mint.decimals)).toString();
```

Comment:
Maybe we can use [amountToUiAmount ](https://solana-labs.github.io/solana-program-library/token/js/functions/amountToUiAmount.html) here. Does the same, adhering to conventions.

Reply:
looks like this does a network call to verify decimals and then do the conversion. is that worth the extra network call?

Reply:
The SDK can be improved in the backend. Using the SDK guarantees compliance & best practices.
If everyone rolled their own math, we're prone to inconsistencies across the ecosystem.